src/applications/proxy-rewrite/proxy-rewrite-entry.jsx

```diff
@@ -170,12 +176,95 @@ function shouldActivateInjectedAssets(whitelistItem, proxyRewriteCookieValue) {
   return true;
 }
 
-if (
-  shouldActivateInjectedAssets(
-    getMatchedWhitelistItem(),
-    getProxyRewriteCookieValue(),
-  )
+function getHostnameOverride() {
+  // default to vagovprod
+  const targetEnvironment = (window.location.search.match(
+    /\btargetEnvironment=(\w+)/,
+  ) || [])[1];
+
+  // localhost is not available in the buckets
+  if (targetEnvironment === 'localhost') {
+    return {
+      targetEnvironment,
+      hostnameOverride: 'http://localhost:3001',
+    };
+  }
+
+  // if the bucket is not found, an empty string will use relative paths
+  return {
+    targetEnvironment,
+    hostnameOverride: BUCKETS[targetEnvironment] || '',
+  };
+}
+
+const scriptPaths = [
+  '/js/settings.js',
+  '/generated/polyfills.entry.js',
+  '/generated/vendor.entry.js',
+  '/generated/proxy-rewrite.entry.js',
+];
+
+const linkPaths = ['/generated/styleConsolidated.css'];
+
+function removeInjectedHeaderFooter(
+  links = linkPaths,
+  scripts = scriptPaths,
+  docHead = document.head,
+) {
+  Array.from(docHead.getElementsByTagName('script'))
+    .filter(node => scripts.includes(node.getAttribute('src')))
+    .forEach(node => docHead.removeChild(node));
+
+  Array.from(docHead.getElementsByTagName('link'))
+    .filter(node => links.includes(node.getAttribute('href')))
+    .forEach(node => docHead.removeChild(node));
+}
+
+function addOverrideHeaderFooter(
+  hostname,
+  links = linkPaths,
+  scripts = scriptPaths,
+  docHead = document.head,
 ) {
-  redirectIfNecessary(window);
-  activateInjectedAssets();
+  const docFragment = document.createDocumentFragment();
+  // add <script>s
+  scripts.forEach(path => {
+    const scriptNode = document.createElement('script');
+    scriptNode.setAttribute('src', `${hostname}${path}`);
+
+    docFragment.appendChild(scriptNode);
+  });
+
+  // add <link>
+  links.forEach(path => {
+    const linkNode = document.createElement('link');
+    linkNode.setAttribute('rel', 'stylesheet');
+    linkNode.setAttribute('href', `${hostname}${path}`);
+
+    docFragment.appendChild(linkNode);
+  });
+
+  docHead.appendChild(docFragment);
 }
+
+function main() {
+  // if a build type is passed in the url, then the header for the specific build type is used
+  const { targetEnvironment, hostnameOverride } = getHostnameOverride();
+
+  if (targetEnvironment && targetEnvironment !== currentEnvironment.BUILDTYPE) {
```

Comment:
This is the main change. If there's no query param set, then the proxy-rewrite behaves as it does currently. If there is an environment set in the query param, then this will
- remove the current header and footer using the existing implementation 
  - the current implementation uses mutation observers which will only fire if the observer is added _before_ the DOM is created 
  - Decided against refactoring the mutation observer approach for something that would work pre and post DOM creation- we can revisit this but I think it's out of the scope of this change. 
- removes the injected header and footer
  - this is new logic- it basically filters the head for the script and link elements related to the teamsite injection
  - there is some inline CSS i didn't swap out that's necessary for making the existing header not visible- I'm not sure why this CSS is inlined but we might look into consolidating it with the main page CSS
- adds the new header / footer tags with the target environment
  - these will have a low priority download but this should only be used for testing purposes 
  - this approach doesn't short circuit the settings, polyfills, or vendor files and unlike removing CSS (a declarative asset that can be reset by swapping it), removing JS doesn't undo imperative changes made to the global state.